core/src/main/java/org/apache/calcite/rex/RexBuilder.java

```diff
@@ -832,6 +858,25 @@ public RexNode makeAbstractCast(RelDataType type, RexNode exp, boolean safe) {
     return new RexCall(type, operator, ImmutableList.of(exp));
   }
 
+  /**
+   * Creates a call to CAST or SAFE_CAST operator with a FORMAT clause.
+   *
+   * @param type Type to cast to
+   * @param exp  Expression being cast
+   * @param safe Whether to return NULL if cast fails
+   * @param format Conversion format for target type
+   * @return Call to CAST operator
+   */
+  public RexNode makeAbstractCast(RelDataType type, RexNode exp, boolean safe, RexLiteral format) {
+    SqlOperator operator =
```

Comment:
operator can be final?

Reply:
Yes that should work, will make the change to the existing `makeAbstractCast` overload too.